data-service/src/main/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDB.java

```diff
@@ -76,4 +83,81 @@ public Environment getEnvironment(
 
     return environmentMapper.toEnvironment(environmentDDBRecord);
   }
+
+  @Override
+  public Environment createEnvironmentTargetVersion(final Environment environment)
+      throws StorageException, EnvironmentExistsException {
```

Comment:
It doesn't look like this code actually ever throws an EnvironmentExistsException?

Reply:
oh yea good catch, forgot to add a use case. That's why integration tests that test every exception are so valuable